Rubberduck.VBEEditor/SafeComWrappers/VB/VBProjectFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using Rubberduck.VBEditor.SafeComWrappers.VB.Abstract;
+using VBAIA = Microsoft.Vbe.Interop;
+using VB6IA = Microsoft.VB6.Interop.VBIDE;
+
+namespace Rubberduck.VBEditor.SafeComWrappers.VB
+{
+    public static class VBProjectFactory
+    {
+        public static IVBProject Create(object project)
+        {
+            switch (project)
+            {
+                case VBAIA.VBProject vbap:
+                    return new VBA.VBProject(vbap);
+
+                case VB6IA.VBProject vb6p:
```

Comment:
nevermind. that's C#7?

Reply:
Yes. Pattern matching or Tuple deconstruction. Depends on how exactly it's implemented...